core/frontend/src/render/webgl/AtmosphereUniforms.ts

```diff
@@ -0,0 +1,218 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+import { Matrix3d, Point3d, Transform } from "@itwin/core-geometry";
+import { Atmosphere } from "@itwin/core-common";
+import { WebGLDisposable } from "./Disposable";
+import { desync, sync, SyncTarget } from "./Sync";
+import { Target } from "./Target";
+import { UniformHandle } from "./UniformHandle";
+import { Matrix3 } from "./Matrix";
+
+export const MAX_SAMPLE_POINTS = 40; // Maximum number of sample points to be used for the in-scattering and out-scattering computations.
+
+export class AtmosphereUniforms implements WebGLDisposable, SyncTarget {
+  private _atmosphere?: Atmosphere.Settings;
+  public get atmosphere(): Atmosphere.Settings | undefined {
+    return this._atmosphere;
+  }
+
+  // Atmosphere effect parameters
+  private _earthCenter = new Point3d();
+  private _earthScaleMatrix = new Matrix3d(new Float64Array([1, 0, 0, 0, 1, 0, 0, 0, 1]));
+  private _inverseEllipsoidRotationMatrix = new Matrix3d(new Float64Array([1, 0, 0, 0, 1, 0, 0, 0, 1]));
+  private _atmosphereRadiusScaleFactor = 1.0;
+  private _atmosphereScaleMatrix = new Matrix3d(new Float64Array([1, 0, 0, 0, 1, 0, 0, 0, 1]));
+  private _atmosphereMaxDensityThresholdScaleFactor = 1.0;
+  private _densityFalloff = 0.0;
+  private _outScatteringIntensity = 1.0;
+  private _exposure = 0.0;
+  private _scatteringCoefficients = new Float32Array(3);
+
+  // Iteration parameters
+  private _numViewRaySamples = 0.0;
+  private _numSunRaySamples = 0.0;
+
+  // utility
+  public syncKey = 0;
+  private _scratchMatrix3d = new Matrix3d();
+
+  public update(target: Target): void {
+    desync(this);
```

Comment:
Should only `desync` if the atmosphere has changed. See `RealityModelUniforms`, `ThematicUniforms`, `ShadowUniforms`, etc for examples.